packages/gatsby/src/query/graphql-runner.js

```diff
@@ -0,0 +1,40 @@
+const { graphql } = require(`graphql`)
+
+const withResolverContext = require(`../schema/context`)
+const { LocalNodeModel } = require(`../schema/node-model`)
+
+class GraphQLRunner {
+  constructor(store) {
+    this.store = store
+    const nodeStore = require(`../db/nodes`)
+    const createPageDependency = require(`../redux/actions/add-page-dependency`)
+    const { schema, schemaCustomization } = this.store.getState()
+
+    this.nodeModel = new LocalNodeModel({
+      nodeStore,
+      schema,
+      schemaComposer: schemaCustomization.composer,
+      createPageDependency,
+    })
+  }
+
+  query(query, context) {
+    const { schema, schemaCustomization } = this.store.getState()
+
+    return graphql(
+      schema,
+      query,
+      context,
+      withResolverContext({
+        schema,
+        schemaComposer: schemaCustomization.composer,
+        context,
+        customContext: schemaCustomization.context,
+        nodeModel: this.nodeModel,
+      }),
+      context
```

Comment:
this is not important but do we know what depends on injecting `context` for root value and variables?

Reply:
I don't think anything depends on root value. I'd fix it in a separate PR, move it so that we pass context and variables separately.